### ISSUE TITLE
WPS500 False-positive with `return` and not `break`

### ISSUE THREAD

**Issue by webknjaz** [wemake-services/wemake-python-styleguide#1533]
This rule wants
```python
needle = None
for smth in it:
    if smth is worthy:
        needle = smth
        break
else:
    raise LookupError
```
But inside of the lookup-style functions it's reasonable to just `return` early instead of maintaining an extra var:
```python
def func(it):
    for smth in it:
        if smth is worthy:
            return smth
    else:
        raise LookupError
```

This case does not trigger `raise` when `return` interrupts the loop, just like `break`. I think it should not be reported.

**Comment by orsinium** [comment]
In case of using return, you can omit `else`:

```python
def func(it):
    for smth in it:
        if smth is worthy:
            return smth
    raise LookupError
```

**Comment by orsinium** [comment]
I hardly can imagine a case when `else` for `for` is needed without `break`. [According to the documentation](https://docs.python.org/3/reference/compound_stmts.html#for), `break` is the only flow control statement that "terminates the loop without executing the else clause’s suite". For every other case, `else` can be just inlined.

If you don't mind, I close the issue.

+ Please, feel free to re-open the issue if the proposed solution doesn't work for you.
+ Please, make a PR if you have an idea of how to make the error message or documentation better.
